webserver/views/api/v1/datasets.py

```diff
@@ -194,7 +224,22 @@ def delete_class(dataset_id):
 
     :resheader Content-Type: *application/json*
     """
-    raise NotImplementedError
+    ds = get_check_dataset(dataset_id)
+    if ds["author"] != current_user.id:
+        raise api_exceptions.APIUnauthorized("You can't delete this class.")
+    class_dict = request.get_json()
+    if "name" not in class_dict:
+        raise api_exceptions.APIBadRequest("name key missing in JSON request.")
+    if class_dict["name"] not in (classes["name"] for classes in ds["classes"]):
+        raise api_exceptions.APIBadRequest("Class does not exists.")
+    try:
+        db.dataset.delete_class(class_dict, dataset_id)
+    except db.exceptions.NoDataFoundException as e:
+        raise api_exceptions.APINotFound(e.message)
+    return jsonify(
+        success=True,
+        message="Class deleted."
```

Comment:
Some APIs return data in similar cases, some don't. Examples:

* https://developer.spotify.com/web-api/unfollow-artists-users/
* https://stripe.com/docs/api#delete_coupon

And if we aren't returning anything it should probably be `204` instead. See http://stackoverflow.com/a/2342589/272770 and references there.